testsuite/MDAnalysisTests/coordinates/test_h5md.py

```diff
@@ -77,336 +75,226 @@ def test_get_writer_2(self, ref, reader, tmpdir):
+        assert_almost_equal(universe.trajectory.ts.dimensions,
+                            [52.807877, 52.807877, 52.807877, 90., 90., 90.],
+                            decimal=self.prec)
+        universe.trajectory[2]
+        assert_almost_equal(universe.trajectory.ts.dimensions,
+                            [52.839806, 52.839806, 52.839806, 90., 90., 90.],
+                            decimal=self.prec)
+
+    def test_h5md_data_step(self, universe):
+        for ts, step in zip(universe.trajectory, (0, 25000, 50000)):
+            assert_equal(ts.data['step'], step)
+
+    def test_rewind(self, universe):
+        universe.trajectory[1]
+        universe.trajectory.rewind()
+        assert universe.trajectory.ts.frame == 0
+
+    def test_next(self, universe):
+        universe.trajectory.rewind()
+        universe.trajectory.next()
+        assert universe.trajectory.ts.frame == 1
+
+    def test_jump_last_frame(self, universe):
+        universe.trajectory[-1]
+        assert universe.trajectory.ts.frame == 2
+
+    @pytest.mark.parametrize("start, stop, step", ((0, 2, 1),
+                                                   (1, 2, 1)))
+    def test_slice(universe, start, stop, step):
+        frames = [universe.trajectory.ts.frame
+                  for ts in universe.trajectory[start:stop:step]]
+        assert_equal(frames, np.arange(start, stop, step))
+
+    @pytest.mark.parametrize("start, stop, step", ((0, 2, 1),
+                                                   (1, 2, 1)))
+    def test_slice(self, universe, start, stop, step):
+        frames = [universe.trajectory.ts.frame
+                  for ts in universe.trajectory[start:stop:step]]
+        assert_equal(frames, np.arange(start, stop, step))
+
+    @pytest.mark.parametrize("array_like", [list, np.array])
+    def test_array_like(self, universe, array_like):
+        array = array_like([0, 2])
+        frames = [universe.trajectory.ts.frame
+                  for ts in universe.trajectory[array]]
+        assert_equal(frames, array)
+
+    def test_list_indices(self, universe):
+        indices = [0, 1, 2, 1, 2, 2, 0]
+        frames = [universe.trajectory.ts.frame
+                  for ts in universe.trajectory[indices]]
+        assert_equal(frames, indices)
+
+    @pytest.mark.parametrize('group, attr', (('position', 'positions'),
+                                             ('velocity', 'velocities'),
+                                             ('force', 'forces')))
+    def test_no_group(self, h5md_file, outfile, attr, group):
         with h5md_file as f:
             with h5py.File(outfile, 'w') as g:
                 f.copy(source='particles', dest=g)
                 f.copy(source='h5md', dest=g)
-                del g['particles/trajectory/force']
-        u = mda.Universe(TPR_xvf, outfile, format='H5MD')
+                del g[f'particles/trajectory/{group}']
+        u = mda.Universe(TPR_xvf, outfile)
         with pytest.raises(NoDataError):
```

Comment:
Wherever possible, can we have a `match` for the associated error/warning we are matching?
This makes sure that we are testing the error we actually want to test (you'd think this wouldn't be needed, but in practice it happens more often than you'd think).

Reply:
I'll go through and add matches